# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Vinod5566773/flyrank-capstone/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [2]:
lane = "Refresh / Content Opportunity Scoring"

assert lane != "", "Please select a research lane."
print("Selected lane:", lane)
print("Goal: prioritize content items for review or refreshing.")

Selected lane: Refresh / Content Opportunity Scoring
Goal: prioritize content items for review or refreshing.


My lane is **Refresh / Content Opportunity Scoring**. I chose this lane because I want to identify content items that may need review or refreshing based on changes in their impressions and recent performance. The goal is to help a content team prioritize which items to review first, rather than manually checking every item. I will use the available data to create a directional, decision-support ranking and evaluate its limitations.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research question:** Which content items should be reviewed first for possible improvement?

**Decision:** Prioritize content items for review based on their observed performance signals.

**Who acts:** A content reviewer examines the recommendations and decides whether to refresh, improve, or monitor each item.

**Cost of a wrong call:** Reviewing a healthy item wastes time, while overlooking a declining item may delay useful improvements.


In [3]:
research_question = "Which content items should be reviewed first?"
decision_maker = "Content reviewer"
wrong_call_cost = "Wasted review time or missed improvement opportunities"

assert research_question
assert decision_maker
assert wrong_call_cost

print("Research question:", research_question)
print("Decision maker:", decision_maker)
print("Cost of a wrong call:", wrong_call_cost)

Research question: Which content items should be reviewed first?
Decision maker: Content reviewer
Cost of a wrong call: Wasted review time or missed improvement opportunities


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import pandas as pd

url = "https://raw.githubusercontent.com/Vinod5566773/flyrank-capstone/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

print("Total rows:", len(df))
print("Declining pages:", (df["trend_direction"] == "down").sum())
print("Pages with 500+ impressions:", (df["impressions_90d"] >= 500).sum())

Total rows: 30000
Declining pages: 16262
Pages with 500+ impressions: 16726


The starter dataset contains 30,000 content items. Of these, 16,262 have a downward trend, and 16,726 have at least 500 impressions over 90 days. These counts show that the dataset contains many items with declining trends and many with substantial visibility. This makes it useful to investigate how content items can be prioritized for review. These counts describe the dataset; they do not prove that refreshing content will improve performance.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This analysis can describe observed patterns in content impressions and trends. It can provide directional, decision-support recommendations about which content items may deserve review first. However, the analysis cannot prove that refreshing a page will cause its performance to improve. It also cannot predict Google's algorithm or guarantee future traffic. Recommendations should be reviewed by a person before action is taken.


In [6]:
claims = [
    "Describe observed patterns in content performance.",
    "Provide directional decision support for content review."
]

limitations = [
    "Cannot prove that refreshing content causes improvement.",
    "Cannot predict Google's algorithm or guarantee future traffic."
]

assert len(claims) == 2
assert len(limitations) == 2

print("Supported claims:")
for claim in claims:
    print("-", claim)

print("\nLimitations:")
for limitation in limitations:
    print("-", limitation)

Supported claims:
- Describe observed patterns in content performance.
- Provide directional decision support for content review.

Limitations:
- Cannot prove that refreshing content causes improvement.
- Cannot predict Google's algorithm or guarantee future traffic.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.